blivet/fstab.py

```diff
@@ -0,0 +1,790 @@
+# fstab.py
+# Fstab management.
+#
+# Copyright (C) 2023  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU Lesser General Public License v.2, or (at your option) any later
+# version. This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY expressed or implied, including the implied
+# warranties of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
+# the GNU Lesser General Public License for more details.  You should have
+# received a copy of the GNU Lesser General Public License along with this
+# program; if not, write to the Free Software Foundation, Inc., 51 Franklin
+# Street, Fifth Floor, Boston, MA 02110-1301, USA.  Any Red Hat trademarks
+# that are incorporated in the source code or documentation are not subject
+# to the GNU Lesser General Public License and may only be used or
+# replicated with the express permission of Red Hat, Inc.
+#
+# Red Hat Author(s): Jan Pokorny <japokorn@redhat.com>
+#
+
+import os
+
+from libmount import Table, Fs, MNT_ITER_FORWARD
+from libmount import Error as LibmountException
+
+import logging
+log = logging.getLogger("blivet")
+
+
+class FSTabOptions(object):
+    """ User prefered fstab settings object intended to be attached to device.format.
+        Set variables override otherwise automatically obtained values put into fstab.
+    """
+
+    def __init__(self):
+        self.freq = None
+        self.passno = None
+
+        # prefered spec identification type; default "UUID"
+        # possible values: None, "UUID", "LABEL", "PARTLABEL", "PARTUUID", "PATH"
+        self.spec_type = None
+
+        # list of fstab options to be used
+        self.mntops = []
+
+
+class FSTabEntry(object):
+    """ One processed line of fstab
+    """
+
+    def __init__(self, spec=None, file=None, vfstype=None, mntops=None,
+                 freq=None, passno=None, comment=None, *, entry=None):
+
+        # Note: "*" in arguments means that every following parameter can be used
+        # only with its key (e.g. "FSTabEntry(entry=Fs())")
+
+        if entry is None:
+            self._entry = Fs()
+        else:
+            self._entry = entry
+
+        if spec is not None:
+            self.spec = spec
+        if file is not None:
+            self.file = file
+        if vfstype is not None:
```

Comment:
For the fstab fields I am using the same names as in man pages for fstab (without `fs_` prefix). I hope this way it will be easier to cross reference the blivet change to the fstab.